sensing/src/test/java/com/google/android/sensing/impl/SensorManagerImplTest.kt

```diff
@@ -0,0 +1,204 @@
+import android.os.Build
+import androidx.lifecycle.testing.TestLifecycleOwner
+import androidx.test.core.app.ApplicationProvider
+import com.google.android.sensing.SensingEngine
+import com.google.android.sensing.SensingEngineConfiguration
+import com.google.android.sensing.SensorManager
+import com.google.android.sensing.model.CaptureInfo
+import com.google.android.sensing.testing.TEST_EXTERNAL_ID
+import com.google.android.sensing.testing.TEST_OUTPUT_FOLDER
+import com.google.android.sensing.testing.TEST_SENSOR_FACTORY
+import com.google.android.sensing.testing.TEST_SENSOR_TYPE
+import com.google.android.sensing.testing.TestApplication
+import com.google.android.sensing.testing.TestSensorCaptureRequest
+import com.google.android.sensing.testing.TestSensorInitConfig
+import com.google.common.truth.Truth
+import kotlin.test.assertFailsWith
+import kotlinx.coroutines.test.runTest
+import org.junit.After
+import org.junit.AfterClass
+import org.junit.Assert.assertThrows
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+/** Unit tests for [SensorManagerImpl]. */
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [Build.VERSION_CODES.P], application = TestApplication::class, manifest = Config.NONE)
+class SensorManagerImplTest {
+  private val sensorManager =
+    SensorManager.getInstance(ApplicationProvider.getApplicationContext<TestApplication>())
+
+  private val sensingEngine =
+    SensingEngine.getInstance(ApplicationProvider.getApplicationContext<TestApplication>())
+
+  @Before
+  fun beforeSetup() {
+    check(
+      ApplicationProvider.getApplicationContext<TestApplication>()
+        is SensingEngineConfiguration.Provider
+    ) {
+      "Few tests require a custom application class that implements SensingEngineConfiguration.Provider"
+    }
+    sensorManager.reset(TEST_SENSOR_TYPE)
+    sensorManager.unregisterSensorFactory(TEST_SENSOR_TYPE)
+  }
+
+  @After
+  fun afterSetup() {
+    sensorManager.reset(TEST_SENSOR_TYPE)
+    sensorManager.unregisterSensorFactory(TEST_SENSOR_TYPE)
+  }
+
+  @Test
+  fun registerSensorFactory_checkRegistration_shouldBeTrue() {
+    sensorManager.registerSensorFactory(TEST_SENSOR_TYPE, TEST_SENSOR_FACTORY)
+
+    Truth.assertThat(sensorManager.checkRegistration(TEST_SENSOR_TYPE)).isTrue()
+  }
+
+  @Test
+  fun registerSensorFactory_reRegister_shouldGiveIllegalStateException() {
+    sensorManager.registerSensorFactory(TEST_SENSOR_TYPE, TEST_SENSOR_FACTORY)
+
+    assertThrows(IllegalStateException::class.java) {
+      sensorManager.registerSensorFactory(TEST_SENSOR_TYPE, TEST_SENSOR_FACTORY)
+    }
+  }
+
+  @Test
+  fun unregisterSensorFactory_checkRegistration_shouldBeFalse() {
+    sensorManager.registerSensorFactory(TEST_SENSOR_TYPE, TEST_SENSOR_FACTORY)
+    Truth.assertThat(sensorManager.checkRegistration(TEST_SENSOR_TYPE)).isTrue()
```

Comment:
nit. Import `Truth.assertThat` so that you can simply write `assertThat` every where.

Reply:
done